kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/KrpcFilter.java

```diff
@@ -11,4 +11,11 @@
  */
 public /* sealed */ interface KrpcFilter {
 
+    /**
+     * Called when the Filter is removed, for instance when the client connection is closed
+     */
+    default void onFilterRemoved() {
```

Comment:
If we keep this we need to update the class comment as its no longer a marker interface...